backend/src/websocket.rs

```diff
@@ -1,20 +1,65 @@
-use actix::{Actor, StreamHandler};
+use actix::{Actor, Addr, AsyncContext, Handler, Message, StreamHandler};
+use actix_web::web;
 use actix_web_actors::ws;
 
+use serde::Serialize;
+
 use super::game::Game;
 use super::game_objects::hex_objects::content::Content;
 use super::game_objects::hex_objects::wall::Wall;
 use super::game_objects::unit::Unit;
+use crate::appstate::AppState;
 
 use super::api;
 
 /// Define http actor
-pub struct Websocket;
+#[derive(Debug)]
+pub struct Websocket {
+    pub self_addr: Option<Addr<Websocket>>,
+    pub app_state: web::Data<AppState>,
+}
+
+#[derive(Message, Debug)]
+#[rtype(result = "()")]
+pub struct Msg(pub String);
```

Comment:
let's think a bit about naming here

and maybe we can hide this into structs from api/response

Reply:
https://github.com/Miol-Mor/battle-game/issues/63